src/BuildingRegistry/Building/Building.cs

```diff
@@ -123,6 +124,35 @@ public void RealizeConstruction()
                 finalPosition,
                 function,
                 hasDeviation));
+
+            AddCommonBuildingUnit(persistentLocalIdGenerator);
+        }
+
+        private void AddCommonBuildingUnit(IPersistentLocalIdGenerator persistentLocalIdGenerator)
+        {
+            if (BuildingStatus != BuildingStatus.Planned
+                && BuildingStatus != BuildingStatus.UnderConstruction
+                && BuildingStatus != BuildingStatus.Realized)
+            {
+                return;
+            }
```

Comment:
This is the same as the valid statuses check in `PlanBuildingUnit`.
Should we not make a guard function that has a specific exception for both?
Not `BuildingUnitCannotBePlannedException` but more like `BuildingHasInvalidStatusException` or something like that.